client/my-sites/pages/page-list.jsx

```diff
@@ -156,7 +156,8 @@ const Pages = createReactClass( {
 	},
 
 	getNoContentMessage() {
-		const { search, translate } = this.props;
+		const { translate } = this.props;
+		const { search, status = 'published' } = this.props.query;
```

Comment:
Thoughts about this:

- We depend on `this.props.query` being an object, otherwise we'll get an error when we try to do this destructuring assignment.
- You mention [in your question](https://github.com/Automattic/wp-calypso/pull/18345/files#r148956867) assigning `query` from `this.props`, then assigning other constants from `query`. I like that approach because if we add a default empty object query, we get some protection:
   ```js
   const noProps = {};
   const { doesntExist = {} } = noProps;
   const { noError } = doesntExist;  // 🎉

  // specifically in this case
  const { query = {}, translate } = this.props;
  const { search, status = 'published' } = query;
  // search === undefined; status === 'published'
   ```
- It would also be nice to add `propTypes` for this:
  ```js
  propTypes: {
    // …
    query: PropTypes.object
  },
  ```
  [You can go pretty bananas with `propTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes), describing the shape of the object in great details, but I think just saying we expect an object is sufficient here 🙂